src/category_theory/abelian/derived.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2022 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca, Adam Topaz
+-/
+
+import category_theory.abelian.homology
+import category_theory.functor.derived
+import category_theory.abelian.projective
+import category_theory.limits.constructions.epi_mono
+
+/-!
+# Zeroth left derived functors
+
+If `F : C ⥤ D` is an additive right exact functor between abelian categories, where `C` has enough
+projectives, we provide the natural isomorphism `(F.left_derived 0) ≅ F`.
+
+## Main definitions
+
+* `category_theory.abelian.functor.left_derived_zero_iso_self`: the natural isomorphism
+  `(F.left_derived 0) ≅ F`.
+
+## Main results
+* `preserves_exact_of_preserves_finite_colimits_of_epi`: if `preserves_finite_colimits F` and
+  `epi g`, then `exact (F.map f) (F.map g)` if `exact f g`.
+
+-/
+
+noncomputable theory
+
+universes w v u
+
+open category_theory.limits category_theory category_theory.functor
+
+variables {C : Type u} [category.{w} C] {D : Type u} [category.{w} D]
+variables (F : C ⥤ D) {X Y Z : C} (f : X ⟶ Y) {g : Y ⟶ Z}
+
+namespace category_theory.preadditive
+
+variables [preadditive C] [preadditive D] [additive F]
+
+/-- The isomorphism `F.obj (cokernel f) ≅ cokernel (F.map f)` if `preserves_finite_colimits F`. -/
+def preserves_cokernel_of_preserves_finite_colimits [preserves_finite_colimits F] [has_cokernel f]
```

Comment:
I didn't find a reasonable file that has the imports one needs to have these results.